vpr/src/noc/noc_traffic_flows.h

```diff
@@ -0,0 +1,307 @@
+#ifndef NOC_TRAFFIC_FLOWS_H
+#define NOC_TRAFFIC_FLOWS_H
+
+/**
+ * @file 
+ * @brief This file defines the NocTrafficFlows class, which contains all
+ * communication betwee routers in the NoC.
+ * 
+ * Overview
+ * ======== 
+ * The NocTrafficFlows class contains all traffic flows in a given 
+ * design. A traffic flow is defined by the t_noc_traffic_flow struct.
+ * Each traffic flow is indexed by a unique id that can be used to
+ * retrieve information about them.  
+ * 
+ * The class also associates traffic flows to their source routers (start point)
+ * and sink routers (end point). This is useful if one wants to find traffic
+ * flows based on just the source or sink router.
+ * 
+ * The routes for the traffic flows are expected to change throughout placement
+ * as routers will be moved throughout the chip. Therefore this class provides
+ * a datastructure to keep track of which flows have been updated (re-routed).
+ * 
+ * Finally, this class also stores all router blocks in the design.
+ * 
+ * This class will be primarily used during 
+ * placement to identify which routers inside the NoC(NocStorage) need to
+ * routed to each other.This is important since the router modules can be moved
+ * around to different tiles on the FPGA device.
+ * 
+ */
+#include <iostream>
+#include <unordered_map>
+#include <unordered_set>
+#include <vector>
+#include "clustered_netlist_fwd.h"
+#include "noc_data_types.h"
+#include "vtr_vector.h"
+#include "echo_files.h"
+#include "vtr_util.h"
+
+/*
+ * Whenever a traffic flow is processed (routed a connection between
+ * 2 routers and updated the relevant information) we need to
+ * some way to track it, so that we don't process it again. The flags below indicate whether a traffic flow was processed or not.
+ */
+constexpr bool PROCESSED = true;
+constexpr bool NOT_PROCESSED = false;
+
+/**
+ * @brief Describes a traffic flow within the NoC, which is the communication
+ * between two routers. THe NocTrafficFlows contains a number of this structure
+ * to describe all the communication happening within the NoC.
+ * 
+ */
+struct t_noc_traffic_flow {
+    /** stores the names of the router blocks communicating within this traffic flow*/
+    std::string source_router_module_name;
```

Comment:
Say can be a partial name as long as it uniquely identifies the logical router

Reply:
done